### ISSUE TITLE
Discussion: Input Error Handling for Store Refactor

### ISSUE THREAD

**Issue by jarmoza** [neurobagel/annotation_tool#304]
Now that we are transitioning to a more data dictionary-oriented version of the annotation tool, we should revisit the types of input file errors we expect encounter and how they should be handled.

The first instance of this check (taken from the previous implementation) is in #303, `setDataTable` in `index-refactor.js`

**Comment by surchs** [comment]
Thanks @jarmoza. What errors do you have in mind? Do you mean validating the input files against some criteria?

**Comment by jarmoza** [comment]
@surchs Validation could definitely be one source of errors though that sounds like something somewhat complex given the BIDS spec. I mostly meant things like empty files or ones with incomplete/errored syntax - for both json and tsv.

**Comment by surchs** [comment]
Agreed, makes sense to me. Let's address this once the refactor is done. The simple cases you mention definitely make sense (empty, impossible to read). For the data dictionary we can now also use json schema validation against our own schema that will cover both pure BIDS and BIDS+Neurobagel data dictionaries (https://github.com/neurobagel/bagelbids/blob/main/bagelbids/dictionary_models.py). We'll have to create the schema file and store it somewhere the annotator can access it first though.

**Comment by surchs** [comment]
> Linked to #49. Might be good to consolidate.

oh, yeah, good point. I think #304 is a duplicate of #49. Let's close this issue and continue tracking #49.
